data/constants/keycodes/extras/keycodes_canadian_french_0.0.1.hjson

```diff
@@ -0,0 +1,407 @@
+            "label": "N",
+        }
+        "KC_M": {
+            "key": "FC_M",
+            "label": "M",
+        }
+        "KC_COMM": {
+            "key": "FC_COMM",
+            "label": ",",
+        }
+        "KC_DOT": {
+            "key": "FC_DOT",
+            "label": ".",
+        }
+        "KC_SLSH": {
+            "key": "FC_EACU",
+            "label": "É",
+        }
+/* Shifted symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+ * │ | │ ! │ " │ / │ $ │ % │ ? │ & │ * │ ( │ ) │ _ │ + │       │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+ * │     │   │   │   │   │   │   │   │   │   │   │   │ ¨ │     │
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┐    │
+ * │      │   │   │   │   │   │   │   │   │   │ : │   │ > │    │
+ * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┴────┤
+ * │    │ » │   │   │   │   │   │   │   │ ' │   │   │          │
+ * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+ */
+        "S(FC_HASH)": {
+            "key": "FC_PIPE",
+            "label": "|",
+        }
+        "S(FC_1)": {
+            "key": "FC_EXLM",
+            "label": "!",
+        }
+        "S(FC_2)": {
+            "key": "FC_DQUO",
+            "label": "\"",
+        }
+        "S(FC_3)": {
+            "key": "FC_SLSH",
+            "label": "/",
+        }
+        "S(FC_4)": {
+            "key": "FC_DLR",
+            "label": "$",
+        }
+        "S(FC_5)": {
+            "key": "FC_PERC",
+            "label": "%",
+        }
+        "S(FC_6)": {
+            "key": "FC_QUES",
+            "label": "?",
+        }
+        "S(FC_7)": {
+            "key": "FC_AMPR",
+            "label": "&",
+        }
+        "S(FC_8)": {
+            "key": "FC_ASTR",
+            "label": "*",
+        }
+        "S(FC_9)": {
+            "key": "FC_LPRN",
+            "label": "(",
+        }
+        "S(FC_0)": {
+            "key": "FC_RPRN",
+            "label": ")",
+        }
+        "S(FC_MINS)": {
+            "key": "FC_UNDS",
+            "label": "_",
+        }
+        "S(FC_EQL)": {
+            "key": "FC_PLUS",
+            "label": "+",
+        }
+        "S(FC_CEDL)": {
+            "key": "FC_DIAE",
+            "label": "¨ (dead)",
+        }
+        "S(FC_SCLN)": {
+            "key": "FC_COLN",
+            "label": ":",
+        }
+        "S(FC_LDAQ)": {
+            "key": "FC_RDAQ",
+            "label": "»",
+        }
+        "S(FC_COMM)": {
+            "key": "FC_QUOT",
+            "label": "'",
+        }
+/* AltGr symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+ * │ \ │ ± │ @ │ £ │ ¢ │ ¤ │ ¬ │ ¦ │ ² │ ³ │ ¼ │ ½ │ ¾ │       │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+ * │     │   │   │ € │   │   │   │   │   │ § │ ¶ │ [ │ ] │     │
```

Comment:
Windows OSK doesn't have a Euro symbol here.

Reply:
Apparently the Euro symbol is a somewhat recent addition to the Canadian French layout — it appears in [Windows 10.0.22598.1](http://kbdlayout.info/kbdca/10.0.22598.1) (which is apparently a preview release of Windows 11 23H2, released in the Canary channel on 7 April 2023), but does not appear in [Windows 10.0.17134.1](http://kbdlayout.info/kbdca/10.0.17134.1) (Windows 10 1803).  No idea whether that symbol appears there in any proper releases of Windows 10 and Windows 11 — kbdlayout.info does not have any intermediate versions between those two.

Reply:
The Euro symbol is indeed not output by AltGr+E on Windows 10 22H2 19045.3155, but KDE has had it for a while.

Reply:
Is it policy to restrict input locale declarations in QMK to only what the most common released versions of Windows can do? If so, I can remove `FR_EURO` from the list.